TypeScript: Missing @types/d3-cloud declaration causes TS7016 errors in word-cloud and company visualizations

## Problem

The d3-cloud module is used in several visualization files but lacks TypeScript type definitions, causing TS7016 errors:

- `apps/web/app/collections/_components/word-cloud.tsx` (line 98)
- `apps/web/charts/analyze/org/company/visualization.tsx` (line 3)
- `apps/web/charts/analyze/repo/company/visualization.tsx` (line 3)

## Error Message

```
error TS7016: Could not find a declaration file for module 'd3-cloud'. 
'/Users/huohao/.openclaw/workspace/ossinsight/node_modules/.pnpm/d3-cloud@1.2.9/node_modules/d3-cloud/build/d3.layout.cloud.js' 
implicitly has an 'any' type.
Try `npm i --save-dev @types/d3-cloud` if it exists or add a new declaration (.d.ts) file containing `declare module 'd3-cloud';`
```

## Impact

- TypeScript compilation fails with 3 errors
- Type safety is compromised (implicit 'any' type)
- May affect Vercel deployment if strict type checking is enabled

## Suggested Fix

Install the official TypeScript definitions:

```bash
pnpm add -D @types/d3-cloud
```

The package exists at version 1.2.9: https://www.npmjs.com/package/@types/d3-cloud

## Alternative

If the types package doesn't work with the current d3-cloud version, add a local type declaration:

```typescript
// apps/web/types/d3-cloud.d.ts
declare module 'd3-cloud' {
  interface CloudLayout {
    size(size: [number, number]): this;
    words(words: any[]): this;
    padding(padding: number | (() => number)): this;
    spiral(spiral: 'archimedean' | 'rectangular'): this;
    on(event: 'end', callback: (words: any[]) => void): this;
    start(): void;
  }
  
  export default function cloud(): CloudLayout;
}
```

--- a/apps/web/package.json
+++ b/apps/web/package.json
@@ -89,6 +89,7 @@
     "@repo/typescript-config": "workspace:*",
     "@svgr/webpack": "^8.1.0",
     "@tailwindcss/typography": "^0.5.10",
+    "@types/d3-cloud": "^1.2.9",
     "@types/lodash": "^4.14.199",
     "@types/luxon": "^3.3.2",
     "@types/mdast": "^4.0.0",

